### PR TITLE
Disable MERGE when using hypertables

### LINKED ISSUE #4930: Add explicit error when MERGE command is used with hypertables
### What problem do we want to solve?

As discussed in https://github.com/timescale/timescaledb/issues/4929, the new [MERGE command](https://www.postgresql.org/docs/current/sql-merge.html) (introduced in PG 15) fails with an error when used with hypertables.

Currently a generic catch-all error is presented which is not only confusing, but straightforward wrong (it will be thrown even if the extension is available and preloaded):

```
ERROR:  invalid INSERT on the root table of hypertable "conditions"
HINT:  Make sure the TimescaleDB extension has been preloaded.
```

Before adding full support for MERGE (which at the moment we don't know how much effort it will require), we want to make sure that we present a properly descriptive error to TimescaleDB users. 

```
ERROR:  The MERGE command does not support hypertables in this version 
HINT:  Check https://github.com/timescale/timescaledb/issues/4929 for more information and current status
```

### PR DESCRIPTION
**Pull request by mfundul**
Fixes #4930

Co-authored-by: Lakshmi Narayanan Sreethar <lakshmi@timescale.com>

### CODE REVIEW

File: test/expected/merge.out [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+(2 rows)
+
+-- Print expected values in the conditions table once conditions_updated is merged into it
+-- If a key exists in both tables, we take average of the temperature measured
+-- average logic here is a mess but it works
+SELECT COALESCE(c.time, cu.time) as time,
+       COALESCE(c.location, cu.location) as location,
+       (COALESCE(c.temperature, cu.temperature) + COALESCE(cu.temperature, c.temperature))/2 as temperature 
+FROM conditions AS c FULL JOIN conditions_updated AS cu
+ON c.time = cu.time AND c.location = cu.location;
```
Comment (sb230132) [inline test/expected/merge.out:97]: May be you can remove this query as its unrelated.

**Review by sb230132 — APPROVED** [review]

File: test/expected/merge.out (hunk 1 above)
Comment (mfundul) [reply test/expected/merge.out:97]: I added @lkshminarayanan 's tests for posterity, so that when MERGE is supported the new functionality is tested.

Comment (sb230132) [reply test/expected/merge.out:97]: Ok, thanks for clarifying.

**Review by fabriziomello — CHANGES_REQUESTED** [review]
Would be nice to also add a test executing MERGE in a regular table to make sure our hooks will not interfere on it

File: test/expected/merge.out [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+-- Test that normal PostgreSQL tables can merge without exceptions
+CREATE TABLE conditions_pg AS SELECT * FROM conditions;
+CREATE TABLE conditions_updated_pg AS SELECT * FROM conditions_updated;
+MERGE INTO conditions_pg c
+USING conditions_updated_pg cu
+ON c.time = cu.time AND c.location = cu.location
+WHEN MATCHED THEN
+UPDATE SET temperature = (c.temperature + cu.temperature)/2
+WHEN NOT MATCHED THEN
+INSERT (time, location, temperature) VALUES (cu.time, cu.location, cu.temperature);
```
Comment (fabriziomello) [inline test/expected/merge.out:147]: Nice. Just add the expected output for those regular tables

File: test/expected/merge.out [hunk 3]
```diff
@@ -0,0 +1,147 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+-- Create conditions table with location and temperature
+DROP TABLE IF EXISTS conditions;
+NOTICE:  table "conditions" does not exist, skipping
```
Comment (fabriziomello) [inline test/expected/merge.out:6]:
```suggestion
```

File: test/expected/merge.out [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+SELECT time, location, 14 as temperature
+FROM generate_series(
+	'2021-01-01 00:00:00',
+    '2021-01-01 00:00:09',
+    INTERVAL '5 seconds'
+  ) as time,
+generate_series(1,4) as location;
+-- Create conditions_updated table with location and temperature
+DROP TABLE IF EXISTS conditions_updated;
+NOTICE:  table "conditions_updated" does not exist, skipping
```
Comment (fabriziomello) [inline test/expected/merge.out:32]:
```suggestion
```
Do we need this drop statement?

**Review by fabriziomello — APPROVED** [review]

**Comment by fabriziomello** [comment]
@mfundul Unfortunately this PR was inadvertently closed by my mistake in another PR and it cannot be reopened because PRs closed when merging are locked forever. Sorry about that!!!

**Pull request closed without merging**